javalin-plugins/javalin-swagger-plugin/src/main/kotlin/io/javalin/openapi/plugin/swagger/SwaggerPlugin.kt

```diff
@@ -35,14 +36,17 @@ open class SwaggerPlugin @JvmOverloads constructor(private val configuration: Sw
             routingPath = app.cfg.routing.contextPath,
             basePath = configuration.basePath
         )
+        /** Register handler for swagger ui */
+        app.get(configuration.uiPath, swaggerHandler, *configuration.roles)
 
-        val swaggerWebJarHandler = SwaggerWebJarHandler(
-            swaggerWebJarPath = configuration.webJarPath
-        )
-
-        app
-            .get(configuration.uiPath, swaggerHandler, *configuration.roles)
-            .get("${configuration.webJarPath}/*", swaggerWebJarHandler, *configuration.roles)
+        /** Register webjar handler if and only if there isn't already a [SwaggerWebJarHandler] at configured route */
+        val routes = app.javalinServlet().matcher.findEntries(HandlerType.GET, "${configuration.webJarPath}/*")
+        if(routes.isEmpty() || (routes.isNotEmpty() && routes.first().handler !is SwaggerWebJarHandler)){
```

Comment:
I think it could be simply replaced with sth like: `routes.none { it is SwaggerWebJarHandler }` 🤔